Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Material/MaterialAssignment.h

```diff
@@ -73,11 +74,16 @@ namespace AZ
             const MaterialAssignmentMap& materials, const MaterialAssignmentId& id);
 
         //! Utility function for generating a set of available material assignments in a model
-        MaterialAssignmentMap GetMaterialAssignmentsFromModel(Data::Instance<AZ::RPI::Model> model);
+        MaterialAssignmentMap GetMaterialAssignmentsFromModel(const Data::Asset<AZ::RPI::ModelAsset> modelAsset);
+
+        //! Get material slot labels from a model
+        MaterialAssignmentLabelMap GetMaterialAssignmentSlotLabelsFromModel(const Data::Asset<AZ::RPI::ModelAsset> modelAsset);
 
         //! Find an assignment id corresponding to the lod and label substring filters
         MaterialAssignmentId FindMaterialAssignmentIdInModel(
-            const Data::Instance<AZ::RPI::Model>& model, const MaterialAssignmentLodIndex lodFilter, const AZStd::string& labelFilter);
+            const Data::Asset<AZ::RPI::ModelAsset> modelAsset,
+            const MaterialAssignmentLodIndex lodFilter,
+            const AZStd::string& labelFilter);
```

Comment:
These 3 functions seem they can be `const`.

Reply:
these utility functions do not belong to a class